var/spack/repos/builtin/packages/root/package.py

```diff
@@ -229,6 +233,7 @@ class Root(CMakePackage):
 
     # Python
     depends_on("python@2.7:", when="+python", type=("build", "run"))
+    depends_on("python@2.7:3.10", when="@:6.26.09 +python", type=("build", "run"))
```

Comment:
Should this be a `conflict` indicating an issue with newer pythons on older ROOT? I'm not sure it makes a difference to the concretizer these days, but semantically it might be more appropriate.

Reply:
Historically, one difference between this syntax and conflicts has been that the spack concretizer would not try older dependency versions and just give up stating that there is a conflict. If you are telling me that this is not an issue anymore with newer spack versions, I can turn those back into conflicts.

Reply:
The new concretizer is not "greedy" like the old one, so I think it should work. The real expert is @alalazo who could perhaps share some insight into best practices here?

Reply:
In general it's better to add "positive" directives like done here. Conflicts add penalties to our minimization and they might trigger more easily weird situations where, to avoid the penalty, Spack turns off some other default.